src/custom/utils/operator/errors/OperatorError.ts

```diff
@@ -38,14 +39,20 @@ export enum ApiErrorCodeDetails {
 export default class OperatorError extends Error {
   name = 'OperatorError'
   type: ApiErrorCodes
-  description: ApiError['description']
+  description: ApiErrorObject['description']
 
-  static async getErrorMessage(response: Response, action: 'create' | 'delete') {
+  // Status 400 errors
+  // https://github.com/gnosis/gp-v2-services/blob/9014ae55412a356e46343e051aefeb683cc69c41/orderbook/openapi.yml#L563
+  static apiErrorDetails = ApiErrorCodeDetails
+
+  private static async _getErrorMessage(response: Response, action: 'create' | 'delete') {
     try {
-      const orderPostError: ApiError = await response.json()
+      const orderPostError: ApiErrorObject = await response.json()
 
       if (orderPostError.errorType) {
-        return ApiErrorCodeDetails[orderPostError.errorType]
+        const errorMessage = OperatorError.apiErrorDetails[orderPostError.errorType]
+        // shouldn't fall through as this error constructor expects the error code to exist but just in case
+        return errorMessage || 'Error type exists but no valid error details found.'
```

Comment:
I think we should fallback to the error name instead of this.
Otherwise, the user will see that instead of for instance "AlreadyCancelled".
Which is not good, but at least it's related to the actual error.

Or is that not the case?

Reply:
i like it. you mean `errorType` not error name right?

Reply:
Yes!

Reply:
9f56130

Reply:
oh god I'm tired i didn't even make this change yet lol

Reply:
OK now: ba0d8e3